### PR TITLE
Remove references to Python 2.7 from docs

### PR DESCRIPTION
**Pull request by ktbarrett**
Addresses some points from #1289.

### CODE REVIEW

File: documentation/source/quickstart.rst [hunk 1]
```diff
@@ -23,12 +23,6 @@ Installation via PIP
 
 Cocotb can be installed by running
 
-.. code-block:: bash
-
-    pip3 install cocotb
```
Comment (cmarqu) [inline documentation/source/quickstart.rst]: I'm not so sure about this one - the fact that we don't support Python 2 anymore does not mean that the user cannot have both versions installed still and needs to get the Python-3-pip via ``pip3``.

File: documentation/source/quickstart.rst [hunk 2]
```diff
@@ -10,7 +10,7 @@ Pre-requisites
 
 Cocotb has the following requirements:
 
-* Python 2.7, Python 3.5+ (recommended)
+* Python 3.5+
```
Comment (imphil) [inline documentation/source/quickstart.rst]: Add a comment here like "(Use cocotb < 1.4 if you need Python 2.7 support.)

Comment (eric-wieser) [reply documentation/source/quickstart.rst]: Is there a permalink to the old docs we could point to as well?

File: documentation/source/quickstart.rst (hunk 1 above)
Comment (ktbarrett) [reply documentation/source/quickstart.rst]: Perhaps I should remove the other leg then to remove any confusion?

File: documentation/source/quickstart.rst (hunk 2 above)
Comment (imphil) [reply documentation/source/quickstart.rst]: yes: https://cocotb.readthedocs.io/en/v1.3.0/

Comment (ktbarrett) [reply documentation/source/quickstart.rst]: It's in the bottom left corner.

**Pull request merged**